app/src/main/java/com/courseproject/tindar/BlankNavViewModel.java

```diff
@@ -0,0 +1,35 @@
+package com.courseproject.tindar;
+
+import androidx.lifecycle.LiveData;
+import androidx.lifecycle.MutableLiveData;
+import androidx.lifecycle.ViewModel;
+
+/**
+ * A simple {@link ViewModel} subclass. This is a view model for the blank nav activity to share data among the activity
+ * and its fragments.
+ */
+public class BlankNavViewModel extends ViewModel {
+
+    /**
+     * user id for the logged in user
+     */
+    private final MutableLiveData<String> userId = new MutableLiveData<>();
```

Comment:
Why is this mutable? I'm currently under the impression that the user shouldn't be able to change this other than logging in or out, and each log in/out should generate a new BlankNavViewModel.

Reply:
Honestly, I am not sure, but the examples I found from Google typically work with live data with a view model, and LiveData class itself is abstract so I had to call MutableLivedata. The below link talks about the advantage of using live data.  

https://developer.android.com/topic/libraries/architecture/livedata